activerecord/test/cases/adapters/sqlite3/schema_test.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+# SQLite3 can have attached databases, which behave the same as a schema in PostgreSQL
+require "cases/helper"
+
+class SQLite3SchemaTest < ActiveRecord::SQLite3TestCase
+
+  class SchemaThing < ActiveRecord::Base
+    self.table_name = "test_schema.things"
+  end
+
+  def setup
+    original_verbose = ActiveRecord::Migration.verbose
+    ActiveRecord::Migration.verbose = false
+    ActiveRecord::Schema.define do

```

Comment:
I would say this goes against the current test suite's standards. You should be able to run tests on any DB by changing `ARCONN` and using `if` statements to not run tests that don't apply to specific adapters. Attaching a DB also doesn't seem to work in CI. Can you refactor this to use the standard test schema?

Reply:
Weird, the whole test suite passed on my rails-dev-box. I will not have time to look into it until tomorrow.

There are plenty of tests that are specific to one adapter like that. I basically copied the `schema_test.rb` file name and basic structure from the postgresql side of things: https://github.com/rails/rails/blob/master/activerecord/test/cases/adapters/postgresql/schema_test.rb, there is also an equivalent file in the mysql side.

I'm not quite sure what is wrong here to be honest. I work a lot by following examples I find in the project that match my need. Could you show me what you expect?

I agree a single set of tests to checks schema's behaviors, with conditions to setup the schema correctly for each databse would be optimal, thats what I do in the gem that found this issue. But I'm not confortable doing a refactor of hundreds of lines (all 3 databases's schema_test files) of test code in this massive test suite.

Reply:
PostgreSQL does create an actual schema for its tests, so in my mind I should do the same. 

I understand that you don't want me to use `ActiveRecord::Schema`. I can look at doing it from the connection object as you ask. That was the first thing I did, but I had the weird issue where `Thing.first` and `Thing.create` would return a `Thing` that had nil as id. This gets solved by using the Schema. I guess it's another bug down in the rabbit hole.

I'll look into it tomorrow.